android/src/main/java/io/branch/rnbranch/RNBranchModule.java

```diff
@@ -445,6 +445,21 @@ public void setIdentity(String identity) {
         branch.setIdentity(identity);
     }
 
+   @ReactMethod
+    public void setIdentityAsync(String identity, Promise promise) {
```

Comment:
you dont need to pass a promise as param.
`Promise.resolve` , `Promise.reject` should work.

Reply:
Didn't we test this and find it was necessary for it to work?

Reply:
can you check if java allows `Promise.resolve`  ( promise with a uppercase P ) if not I think we are good.

Reply:
I think e started testing but couldn’t fully run that day. Yaa if it is necessary, lets keep it.